src/middlewared/middlewared/job.py

```diff
@@ -254,6 +254,10 @@ def check_pipe(self, pipe):
         if getattr(self.pipes, pipe) is None:
             raise ValueError("Pipe %r is not open" % pipe)
 
+    def check(self, exception=CallError):
```

Comment:
Would you mind we remove this for now and keep the old way?

I had other plans to implement this, combining wait and error.